### ISSUE TITLE
[classification] Currently using the test set predictions to derive alternative cutoff points

### ISSUE THREAD

**Issue by shane-kercheval** [shane-kercheval/r-predictive-analysis-template#17]
"it is important, especially for small sample sizes, to use an independent data set to derive the cutoff. 

if the training set predictions are used, there is likely a large optimistic bias in the class probabilities that will lead to inaccurate assessments of the sensitivity and specificity

if the test set is used, it is no longer an unbiased source to judge model performance" - APM pg 425

think of how to handle

at the very least, make a note in the final report

**Comment by shane-kercheval** [comment]
perhaps could do a form of k-fold cross validation on the training set (each time we build the initial model) and average(?) the ideal cutoff calculated. We could then store this as part of the object or save it somehow (i think we do this for pre-processing objects, for example).

**Comment by shane-kercheval** [comment]
i ended up doing cross validation method on the training set (using the same folds as the training, and predicting on the 'test' set for each fold) and then taking the mean cutoff value. I'm not sure this is the ideal way in general, but for this project, I think it is a good middle ground to using the raw training/test set. When working with the final models, and assuming there is enough data, I would probably use an extra validation set.
